h2o-core/src/main/java/water/api/RequestServer.java

```diff
@@ -80,6 +80,16 @@
   public static ArrayList<Route> routes() { return routesList; }
   public static Route lookupRoute(RequestUri uri) { return routesTree.lookup(uri, null); }
 
+  private static HttpLogFilter[] _filters=null;
+  public static void setFilters(HttpLogFilter... filters) {
+    _filters=filters;
+  }
+
+  static {
```

Comment:
I am not sure what the source of the objection is (there could be several), but perhaps one solution is a bit that is flipped that prevents filters from being set after registration is finalized (similar to the _acceptRequests bit used by jetty)

If it is some other issue, may you please specify?